src/energyplus/ForwardTranslator.cpp

```diff
@@ -358,49 +358,26 @@ namespace energyplus {
     std::sort(shadingControls.begin(), shadingControls.end(), WorkspaceObjectNameLess());
     std::map<Handle, ShadingControlVector> zoneHandleToShadingControlVectorMap;
     for (auto& shadingControl : shadingControls) {
-      std::set<Handle> thisZoneHandleSet;
       for (auto& subSurface : shadingControl.subSurfaces()) {
         boost::optional<Space> space = subSurface.space();
         if (space) {
           boost::optional<ThermalZone> thermalZone = space->thermalZone();
           if (thermalZone) {
             Handle zoneHandle = thermalZone->handle();
-            if (thisZoneHandleSet.empty()) {
-              // first thermal zone, no clone
-              thisZoneHandleSet.insert(zoneHandle);
-              auto it = zoneHandleToShadingControlVectorMap.find(zoneHandle);
-              if (it == zoneHandleToShadingControlVectorMap.end()) {
-                zoneHandleToShadingControlVectorMap.insert(std::make_pair(zoneHandle, std::vector<ShadingControl>()));
-              }
-              it = zoneHandleToShadingControlVectorMap.find(zoneHandle);
-              OS_ASSERT(it != zoneHandleToShadingControlVectorMap.end());
-              it->second.push_back(shadingControl);
-              shadingControl.additionalProperties().setFeature("Shading Control Sequence Number", (int)it->second.size());
-            } else if (thisZoneHandleSet.find(zoneHandle) != thisZoneHandleSet.end()) {
-              // already in here, good to go
-
-            } else {
-              // additional thermal zone, must clone
-              thisZoneHandleSet.insert(zoneHandle);
-              ShadingControl clone = shadingControl.clone(model).cast<ShadingControl>();
-              // assign clone to control subSurface
-              clone.addSubSurface(subSurface);
-              auto it = zoneHandleToShadingControlVectorMap.find(zoneHandle);
-              if (it == zoneHandleToShadingControlVectorMap.end()) {
-                zoneHandleToShadingControlVectorMap.insert(std::make_pair(zoneHandle, std::vector<ShadingControl>()));
-              }
-              it = zoneHandleToShadingControlVectorMap.find(zoneHandle);
-              OS_ASSERT(it != zoneHandleToShadingControlVectorMap.end());
-              it->second.push_back(clone);
-              clone.additionalProperties().setFeature("Shading Control Sequence Number", (int)it->second.size());
+            auto it = zoneHandleToShadingControlVectorMap.find(zoneHandle);
+            if (it == zoneHandleToShadingControlVectorMap.end()) {
+              zoneHandleToShadingControlVectorMap.insert(std::make_pair(zoneHandle, std::vector<ShadingControl>()));
             }
+            it = zoneHandleToShadingControlVectorMap.find(zoneHandle);
+            OS_ASSERT(it != zoneHandleToShadingControlVectorMap.end());
+            it->second.push_back(shadingControl);
+            shadingControl.additionalProperties().setFeature("Shading Control Sequence Number", (int)it->second.size());
```

Comment:
The FT change seems to make sense though. What's weird and not handled (I have no clue how to properly handle that though) is the case where you have something like:

* Shading Control 1 controls Zone1-SurfaceA1
* Shading Control 2 controls Zone2-SurfaceA2 and Zone2-SurfaceA2

Then what happens to the Shading Control Sequence Number? Shading Control 2 would ahve a Shading Control Sequence Number of 2 (assuming it wasn't renamed, since the sort is done on name...), but Zone2-SurfaceA2 is referenced by only one shading control with a sequence number of 2. Does that make sense (I genuinely don't know).


@shorowit thoughts please? It's been a while I've used ShadingControls; and never since the E+ 9.4 changes, and I don't think the E+ API got any clearer in this specific case...

Reply:
I have no experience with the Shading Control Sequence Numbers, sorry. (I've never used the ShadingControl object in a way where the sequence numbers mattered.)